File: src/frontend/src/app/model/income.ts
export class Income {

  incomeSource: string;
  incomeAmount: number;
  paymentDate: string;
  id: number;
  recurringIncome: boolean
  durationOfRecurrence: number


  constructor(incomeSource: string, incomeAmount: number, paymentDate: string, recurringIncome: boolean,
              durationOfRecurrence: number) {
    this.incomeSource = incomeSource;
    this.incomeAmount = incomeAmount;
    this.paymentDate = paymentDate;
    this.recurringIncome = recurringIncome;
    this.durationOfRecurrence = durationOfRecurrence;
  }
}

File: src/main/java/com/mm/moneymanager/model/Income/Income.java
package com.mm.moneymanager.model.Income;

import com.mm.moneymanager.model.Audit.DateAudit;
import com.mm.moneymanager.model.user.User;
import lombok.*;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;

@EqualsAndHashCode(callSuper = false)
@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Income extends DateAudit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Version
    private Long version;

    private BigDecimal incomeAmount;

    private String incomeSource;

    private Boolean recurringIncome;

    private Boolean incomeArchived;

    private Integer durationOfRecurrence;

    private LocalDate paymentDate;

    @ToString.Exclude
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}
